src/translate_c.zig

```diff
@@ -29,6 +29,10 @@ const Scope = struct {
     id: Id,
     parent: ?*Scope,
 
+    /// When the scope corresponds to a function, keep track of the return type
+    /// so that the return expression can be cast, if necessary
+    return_type: ?clang.QualType = null,
+
```

Comment:
This should go in a `Scope.Block` and you should add a function to find the first block where this is not null.